src/main/scala/com/yotpo/metorikku/input/kafka/KafkaInput.scala

```diff
@@ -1,12 +1,24 @@
 package com.yotpo.metorikku.input.kafka
 
+import java.util.Properties
+
 import com.yotpo.metorikku.input.Reader
-import org.apache.spark.sql.DataFrame
+import com.yotpo.metorikku.instrumentation.KafkaLagWriter
 import com.yotpo.metorikku.session.Session.getSparkSession
+import org.apache.kafka.clients.consumer.KafkaConsumer
+import org.apache.spark.sql.{DataFrame, SparkSession}
 
-case class KafkaInput(name: String, servers: Seq[String], topic: String,
+case class KafkaInput(name: String, servers: Seq[String], topic: String, consumerGroup: Option[String],
                       options: Option[Map[String, String]]) extends Reader {
   def read(): DataFrame = {
+    consumerGroup match {
+      case Some(group) =>
+        val consumer = createKafkaConsumer(group)
+        val lagWriter = new KafkaLagWriter(consumer, topic)
+        val ss = SparkSession.builder().getOrCreate()
```

Comment:
It's a hidden dependency, it will be hard to test this because of this.
Is it hard to get it from the outside?

Reply:
👍